src/user-activity/user-activity.service.ts

```diff
@@ -0,0 +1,157 @@
+import { UsersService } from 'src/users/users.service';
+import { Injectable } from '@nestjs/common';
+import { InjectRepository } from '@nestjs/typeorm';
+import { Repository } from 'typeorm';
+import { actionType, UserActivity } from './entities/userActivity.entity';
+import UserActivitySave from './dto/user-activity-save.input';
+import { User } from 'src/users/entities/user.entity';
+import { Event } from 'src/events/entities/event.entity';
+@Injectable()
+export class UserActivityService {
+  constructor(
+    @InjectRepository(UserActivity)
+    private readonly userActivityRepository: Repository<UserActivity>,
+    private readonly usersService: UsersService,
+  ) {}
+
+  saveParticipantActivity(type, user, event) {
+    //Interested
+    let action = actionType.Interested;
+    let score = 2;
+    let weight = 0.03;
+    if (type == 2) {
+      //Take part
+      action = actionType.TakePart;
+      score = 3;
+      weight = 0.05;
+    }
+    this.createOrUpdate(user, action, event.type, score, weight);
+
+    this.createOrUpdate(user, actionType.Category, event.type, 1, 0.03);
```

Comment:
Dlaczego dla wagi nie korzystasz ze zmiennej?

Reply:
Bo to i tak nie ostateczna wersja, to się zmieni pewnie jeszcze raz